cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -561,10 +561,14 @@ func isAllowedFlag(flagName string) bool {
 	allowedFlags := sets.New(kubeadmcmdoptions.CfgPath,
 		kubeadmcmdoptions.IgnorePreflightErrors,
 		kubeadmcmdoptions.DryRun,
+		kubeadmcmdoptions.ForceReset,
 		kubeadmcmdoptions.KubeconfigPath,
 		kubeadmcmdoptions.NodeName,
 		kubeadmcmdoptions.KubeconfigDir,
 		kubeadmcmdoptions.UploadCerts,
+		kubeadmcmdoptions.CertificatesDir,
+		kubeadmcmdoptions.NodeCRISocket,
+		kubeadmcmdoptions.CleanupTmpDir,
```

Comment:
Why do we need this change?  We expect users to use a config file rather than cli flags.

Reply:
those are the CLI flags, but now we should support them in the config file as well, we must allow this mix-configuration here, and the set by the CLI flag override the value in the config file.

Reply:
previously, they are supported by the CLI flags.

Reply:
in units somewhere we had a function to disallow the mixture of flags and --config. we should ideally do this for reset

Reply:
the only reason would be to migrate users away from flags. they are unversioned and difficult to deprecate and remove.

Reply:
> we can look at existing commands in kubeadm and follow existing practices.

iirc, there is no clear explanation for why some of them are not allowed the mix configuration (flag and config file), or not clear for me what's the problem will be caused by the mix-configuration here.

I am personally fine with either of them, but if disallow of the mix configuration, we'd better let others know the reason under the hood, or at least we should doc this somewhere.

@pacoxu @SataQiu  @neolit123  pls vote on this, let's see if we can get consensus on this.

Reply:
in order to not confuse the end-user, once the config is set by both flag and config file, we should throw some warning and say which one is effective in the end.

Reply:
Have talked with @pacoxu on the slack for this,  he suggests me to pull this out in another pr and we can continue the discussion there instead of block this.

And also, seems like @SataQiu @neolit123 doesn't like the mix configuration too much, I am going to respin and revert the change here back.

Reply:
track it here instead: https://github.com/kubernetes/kubeadm/issues/2902